gpustack/routes/inference_backend.py

```diff
@@ -846,3 +868,35 @@ async def update_inference_backend_from_yaml(  # noqa: C901
         raise InternalServerErrorException(
             message=f"Failed to update inference backend from YAML: {e}"
         )
+
+
+def ensure_single_custom_suffix(version: Optional[str]) -> Optional[str]:
+    """
+    Ensure the version name ends with exactly one "-custom".
+    - If missing, append a single "-custom".
+    - If multiple suffixes are present, collapse them to one.
+    - If already has one, leave it unchanged.
+    """
+    if not version:
+        return version
+    # Remove all trailing "-custom" suffixes, then append exactly one
+    base = re.sub(r'(?:-custom)+$', '', version)
+    return f"{base}-custom"
+
+
+def normalize_versions_for_custom_backend(
+    version_configs: Optional[VersionConfigDict],
+) -> Optional[VersionConfigDict]:
+    """
+    Normalize version keys for custom backends: ensure each key ends with a single
+    "-custom", preventing duplicate keys from overwriting.
+    """
+    if not version_configs or not version_configs.root:
+        return version_configs
+
+    new_root: Dict[str, VersionConfig] = {}
+    for version, cfg in version_configs.root.items():
+        normalized = ensure_single_custom_suffix(version)
```

Comment:
I am silly here, why do we need to "ensure" version name ends with exactly one "-custom"? is the customized version not clear with a `-custom` suffix?